src/Microsoft.DotNet.Compiler.Common/BindingRedirectGenerator.cs

```diff
@@ -179,33 +179,40 @@ internal static AssemblyRedirect[] CollectRedirects(IEnumerable<AssemblyReferenc
         {
-            using (var peReader = new PEReader(File.OpenRead(arg.ResolvedPath)))
+            try
             {
-                var metadataReader = peReader.GetMetadataReader();
+                using (var peReader = new PEReader(File.OpenRead(arg.ResolvedPath)))
+                {
+                    var metadataReader = peReader.GetMetadataReader();
 
-                var definition = metadataReader.GetAssemblyDefinition();
+                    var definition = metadataReader.GetAssemblyDefinition();
 
-                var identity = new AssemblyIdentity(
-                    metadataReader.GetString(definition.Name),
-                    definition.Version,
-                    metadataReader.GetString(definition.Culture),
-                    GetPublicKeyToken(metadataReader.GetBlobBytes(definition.PublicKey))
-                );
+                    var identity = new AssemblyIdentity(
+                        metadataReader.GetString(definition.Name),
+                        definition.Version,
+                        metadataReader.GetString(definition.Culture),
+                        GetPublicKeyToken(metadataReader.GetBlobBytes(definition.PublicKey))
+                        );
 
-                var references = new List<AssemblyIdentity>(metadataReader.AssemblyReferences.Count);
+                    var references = new List<AssemblyIdentity>(metadataReader.AssemblyReferences.Count);
 
-                foreach (var assemblyReferenceHandle in metadataReader.AssemblyReferences)
-                {
-                    var assemblyReference = metadataReader.GetAssemblyReference(assemblyReferenceHandle);
-                    references.Add(new AssemblyIdentity(
-                        metadataReader.GetString(assemblyReference.Name),
-                        assemblyReference.Version,
-                        metadataReader.GetString(assemblyReference.Culture),
-                        GetPublicKeyToken(metadataReader.GetBlobBytes(assemblyReference.PublicKeyOrToken))
-                    ));
-                }
+                    foreach (var assemblyReferenceHandle in metadataReader.AssemblyReferences)
+                    {
+                        var assemblyReference = metadataReader.GetAssemblyReference(assemblyReferenceHandle);
+                        references.Add(new AssemblyIdentity(
+                            metadataReader.GetString(assemblyReference.Name),
+                            assemblyReference.Version,
+                            metadataReader.GetString(assemblyReference.Culture),
+                            GetPublicKeyToken(metadataReader.GetBlobBytes(assemblyReference.PublicKeyOrToken))
+                            ));
+                    }
 
-                return new AssemblyReferenceInfo(identity, references.ToArray());
+                    return new AssemblyReferenceInfo(identity, references.ToArray());
+                }
+            }
+            catch (Exception e)
```

Comment:
No idea what PEReader might throw. I made it general to indicate: If anything crashes while reading the assembly info, we'll wrap over the arbitrary exception with a more clear message: "Could not read assembly metadata ... "